Spigot-Server-Patches/0599-Fixed-kick-event-leave-message-not-being-sent.patch

```diff
@@ -0,0 +1,71 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sat, 14 Nov 2020 09:36:43 -0800
+Subject: [PATCH] Fixed kick event leave message not being sent
+
+
+diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
+index 563ae7355f4f4645d21f2fbd494a5b5656c5e664..7b44f231f03d41d83c7680cfa9efc729e8d646ed 100644
+--- a/src/main/java/net/minecraft/server/PlayerConnection.java
++++ b/src/main/java/net/minecraft/server/PlayerConnection.java
+@@ -303,7 +303,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
+         this.networkManager.sendPacket(new PacketPlayOutKickDisconnect(ichatbasecomponent), (future) -> {
+             this.networkManager.close(ichatbasecomponent);
+         });
+-        this.a(ichatbasecomponent); // CraftBukkit - fire quit instantly
++        this.onAllDisconnect(ichatbasecomponent, event.getLeaveMessage()); // CraftBukkit - fire quit instantly // Paper
+         this.networkManager.stopReading();
+         MinecraftServer minecraftserver = this.minecraftServer;
+         NetworkManager networkmanager = this.networkManager;
+@@ -1658,6 +1658,11 @@ public class PlayerConnection implements PacketListenerPlayIn {
+ 
+     @Override
+     public void a(IChatBaseComponent ichatbasecomponent) {
++        // Paper start
++        this.onAllDisconnect(ichatbasecomponent, this.minecraftServer.getPlayerList().disconnect(this.player, "\u00A7e" + this.player.getName() + " left the game"));
++    }
++
++    private void onAllDisconnect(IChatBaseComponent ichatbasecomponent, String quitMessage) {
+         // CraftBukkit start - Rarely it would send a disconnect line twice
+         if (this.processedDisconnect) {
+             return;
+@@ -1673,7 +1678,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
+         */
+ 
+         this.player.p();
+-        String quitMessage = this.minecraftServer.getPlayerList().disconnect(this.player);
++        // String quitMessage = this.minecraftServer.getPlayerList().disconnect(this.player); // Paper
+         if ((quitMessage != null) && (quitMessage.length() > 0)) {
+             this.minecraftServer.getPlayerList().sendMessage(CraftChatMessage.fromString(quitMessage));
+         }
+@@ -1688,7 +1693,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
+             PlayerConnection.LOGGER.info("Stopping singleplayer server as player logged out");
+             this.minecraftServer.safeShutdown(false);
+         }
+-
++        // Paper end
```

Comment:
I meant move this under the declaration, i.e.:
```java
private void onAllDisconnect(IChatBaseComponent ichatbasecomponent, String quitMessage) {
    // Paper end
```

Reply:
oh, ok I get it. I fixed it now.